src/main/kotlin/gg/ingot/iron/sql/controller/Controller.kt

```diff
@@ -46,7 +48,40 @@ sealed interface Controller {
      * @return The prepared statement.
      * @since 1.0
      */
-    fun prepare(@Language("SQL") statement: String, vararg values: Any): ResultSet?
+    fun prepare(@Language("SQL") statement: String, vararg values: Any?): ResultSet?
+
+    /**
+     * Prepares a statement on the database. This method should be preferred over [execute] for security reasons.
+     * @param statement The statement to prepare on the database. This statement should contain named placeholders for
+     * the values, any values passed in through this parameter is not sanitized.
+     * @param values The values to bind to the statement.
+     * @return The prepared statement.
+     * @since 1.3
+     */
+    fun prepare(@Language("SQL") statement: String, values: SqlParameters): ResultSet? {
```

Comment:
Eager method replaces any :<variable> variables in situations it shouldn't, like literal quotes, this also doesn't account for other DBMS'